Run command broken by the introduction of dynamic client

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**The run command crashes with the latest version from master**. This is the generated stack trace:

./sonobuoy run
Running plugins: e2e, systemd-logs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1656011]

goroutine 1 [running]:
github.com/heptio/sonobuoy/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetResourceVersion(0x0, 0x0, 0x3852930)
```        /Users/cosmin/go/src/github.com/heptio/sonobuoy/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:243 +0x41
github.com/heptio/sonobuoy/vendor/k8s.io/apimachinery/pkg/api/meta.resourceAccessor.ResourceVersion(0x1fdce40, 0x0, 0x0, 0x0, 0x0, 0xc4204c5a80)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/vendor/k8s.io/apimachinery/pkg/api/meta/meta.go:355 +0x81
github.com/heptio/sonobuoy/pkg/dynamic.(*APIHelper).ResourceVersion(0xc4200c2d20, 0x0, 0x0, 0x1fd4f00, 0xc420020120, 0x0)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/pkg/dynamic/client.go:97 +0x4a
github.com/heptio/sonobuoy/pkg/client.(*SonobuoyClient).Run(0xc4200c2d50, 0xc4204e2ba0, 0x279a958, 0x0)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/pkg/client/run.go:72 +0x3f6
github.com/heptio/sonobuoy/cmd/sonobuoy/app.submitSonobuoyRun(0xc420122280, 0x279a958, 0x0, 0x0)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/cmd/sonobuoy/app/run.go:106 +0x27a
github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra.(*Command).execute(0xc420122280, 0x279a958, 0x0, 0x0, 0xc420122280, 0x279a958)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x276eaa0, 0x0, 0xc, 0x1c)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra.(*Command).Execute(0x276eaa0, 0xc420091f78, 0xc42004c0b8)
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /Users/cosmin/go/src/github.com/heptio/sonobuoy/main.go:27 +0x2d
```

**What did you expect to happen:**

The run command should start successfully. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T15:22:13Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.10-gke.0", GitCommit:"16ebd0de8e0ab2d1ef86d5b16ab1899b624a77cd", GitTreeState:"clean", BuildDate:"2018-03-20T20:21:01Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}
```
- Sonobuoy version: 
- Kubernetes version: (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`): 
- Sonobuoy tarball (which contains * below)

cc @chuckha

--- a/pkg/client/run.go
+++ b/pkg/client/run.go
@@ -68,8 +68,8 @@ func (c *SonobuoyClient) Run(cfg *RunConfig) error {
 			return errors.Wrap(err, "could not get object namespace")
 		}
 		// err is used to determine output for user; but first extract resource
-		newObj, err := c.dynamicClient.CreateObject(obj)
-		resource, err2 := c.dynamicClient.ResourceVersion(newObj)
+		_, err = c.dynamicClient.CreateObject(obj)
+		resource, err2 := c.dynamicClient.ResourceVersion(obj)
 		if err2 != nil {
 			return errors.Wrap(err, "could not get resource of object")
 		}

--- a/pkg/dynamic/client.go
+++ b/pkg/dynamic/client.go
@@ -94,5 +94,9 @@ func (a *APIHelper) Namespace(obj *unstructured.Unstructured) (string, error) {
 
 // ResourceVersion returns the resource version of a kubernetes object.
 func (a *APIHelper) ResourceVersion(obj *unstructured.Unstructured) (string, error) {
-	return a.Accessor.ResourceVersion(obj)
+	restMapping, err := a.Mapper.RESTMapping(obj.GroupVersionKind().GroupKind(), obj.GroupVersionKind().Version)
+	if err != nil {
+		return "", errors.Wrap(err, "could not get restMapping")
+	}
+	return restMapping.Resource.GroupResource().Resource, nil
 }

